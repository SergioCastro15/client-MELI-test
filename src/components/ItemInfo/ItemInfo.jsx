import React, { useState, useEffect, useContext } from 'react'
import searchContext from '../../context/context'
import "./ItemInfo.scss"

export const ItemInfo = ({ match }) => {

    const { getCurrentItem } = useContext(searchContext)
    const [item, setItem] = useState([])

    useEffect(() => {
        fetchData()
    }, [])

    const fetchData = async () => {
        const data = await getCurrentItem(match.params.id)
        setItem(data)
    }

    return (
        <>
            <div className="container-box">
                <div className="item-info-container">
                    <img className="item-info-picture" src={item.picture} alt=""/>
                    <div className="item-info-text">
                        {
                            item && 
                            <>
                                <p>{item.condition}</p>
                                <h2>{item.item?.title}</h2>
                                <p className="item-price">$ {item.item?.price?.amount}</p>
                                <button className="item-info-button">Comprar</button>
                            </>
                        }
                    </div>
                </div>

                <div className="description-container">
                    <h2 className="text-description-title ">Descripcion del producto</h2>
                    <p className="text-description">{item.description}</p>
                </div>
            </div>
        </>
    )
}
