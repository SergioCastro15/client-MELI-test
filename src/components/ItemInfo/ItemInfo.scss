@import '../../scss/variables';


.container-box{
    background: $white;
    width: 60%;
    margin: auto;

    .item-info-container {
        margin-top: $second-spacing;
        display: flex;
    
        .item-info-picture {
            width: 60%;
        }
    
        .item-info-text {
            width: 40%;
            text-align: initial;

            .item-price {
                font-size: 32px;
            }
    
            .item-info-button {
                background: $blue;
                border: none;
                color: $white;
                width: 200px;
                height: 35px;
            }
        }
    }

    .description-container {
        margin: 20px;
        padding: 20px;

        .text-description-title {
            font-size: 32px;
            text-align: start;
        }

        .text-description {
            text-align: start;
            font-size: 12px;
            color: $gray;
        }
    }

}

@media (max-width: 768px) {
    .container-box{
        width: 90%;
        margin-top: 16px;
    
        .item-info-container {
            display: inline-block;
            .item-info-picture {
                width: 90%;
            }
        
            .item-info-text {
                width: 100%;
                font-size: 10px;
                text-align: center;
        
                .item-info-button {
                    background: $blue;
                    border: none;
                    color: $white;
                    width: 250px;
                    height: 35px;
                }
            }
        }
    
        .description-container {
            .text-description-title {
                font-size: 16px;
                text-align: start;
            }
    
            .text-description {
                text-align: start;
                font-size: 10px;
            }
        }
    
    }
}