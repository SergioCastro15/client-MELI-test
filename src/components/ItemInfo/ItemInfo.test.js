import React from 'react'
import { render } from '@testing-library/react'
import { ItemInfo } from './ItemInfo'
import SearchState from '../../context/state'
import SearchContext from '../../context/context'

import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

const renderWithContext = (component) => {
    const history = createMemoryHistory()

    return {
        ...render(
            <SearchState value={SearchContext}>
                <Router history={history}>
                    {component}
                </Router>
            </SearchState>
        ),history
    }
}

it('should take a snapshot', () => {
    const { asFragment } = renderWithContext(<ItemInfo />)
    
    expect(asFragment(<ItemInfo />)).toMatchSnapshot()
})