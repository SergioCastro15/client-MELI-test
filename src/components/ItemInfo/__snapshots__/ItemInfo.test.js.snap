// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-box"
  >
    <div
      class="item-info-container"
    >
      <img
        alt=""
        class="item-info-picture"
      />
      <div
        class="item-info-text"
      >
        <p />
        <h2 />
        <p
          class="item-price"
        >
          $ 
        </p>
        <button
          class="item-info-button"
        >
          Comprar
        </button>
      </div>
    </div>
    <div
      class="description-container"
    >
      <h2
        class="text-description-title "
      >
        Descripcion del producto
      </h2>
      <p
        class="text-description"
      />
    </div>
  </div>
</DocumentFragment>
`;
